#! /bin/sh /usr/share/dpatch/dpatch-run
## 101_gtk_marshal.dpatch by Bryce Harrington <bryce@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix build failure with gtk+-2.13.x (LP: #238217)

@DPATCH@

diff -Nurp patched/src/widgets/dash-selector.cpp working/src/widgets/dash-selector.cpp
--- patched/src/widgets/dash-selector.cpp	2009-04-07 22:27:49.000000000 -0700
+++ working/src/widgets/dash-selector.cpp	2009-04-07 22:28:31.000000000 -0700
@@ -93,7 +93,7 @@ sp_dash_selector_class_init (SPDashSelec
 					   (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
 					   G_TYPE_FROM_CLASS (klass),
 					   GTK_SIGNAL_OFFSET (SPDashSelectorClass, changed),
-					   gtk_marshal_NONE__NONE,
+					   g_cclosure_marshal_VOID__VOID,
 					   GTK_TYPE_NONE, 0);
 }
 
diff -Nurp patched/src/widgets/font-selector.cpp working/src/widgets/font-selector.cpp
--- patched/src/widgets/font-selector.cpp	2009-04-07 22:27:49.000000000 -0700
+++ working/src/widgets/font-selector.cpp	2009-04-07 22:28:31.000000000 -0700
@@ -134,7 +134,7 @@ static void sp_font_selector_class_init(
                                            GTK_RUN_FIRST,
                                            GTK_CLASS_TYPE(object_class),
                                            GTK_SIGNAL_OFFSET(SPFontSelectorClass, font_set),
-                                           gtk_marshal_NONE__POINTER,
+                                           g_cclosure_marshal_VOID__POINTER,
                                            GTK_TYPE_NONE,
                                            1, GTK_TYPE_POINTER);
 
diff -Nurp patched/src/widgets/gradient-selector.cpp working/src/widgets/gradient-selector.cpp
--- patched/src/widgets/gradient-selector.cpp	2009-04-07 22:27:49.000000000 -0700
+++ working/src/widgets/gradient-selector.cpp	2009-04-07 22:28:31.000000000 -0700
@@ -86,25 +86,25 @@ sp_gradient_selector_class_init (SPGradi
 					    (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
 					    GTK_CLASS_TYPE(object_class),
 					    GTK_SIGNAL_OFFSET (SPGradientSelectorClass, grabbed),
-					    gtk_marshal_NONE__NONE,
+					    g_cclosure_marshal_VOID__VOID,
 					    GTK_TYPE_NONE, 0);
 	signals[DRAGGED] =  gtk_signal_new ("dragged",
 					    (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
 					    GTK_CLASS_TYPE(object_class),
 					    GTK_SIGNAL_OFFSET (SPGradientSelectorClass, dragged),
-					    gtk_marshal_NONE__NONE,
+					    g_cclosure_marshal_VOID__VOID,
 					    GTK_TYPE_NONE, 0);
 	signals[RELEASED] = gtk_signal_new ("released",
 					    (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
 					    GTK_CLASS_TYPE(object_class),
 					    GTK_SIGNAL_OFFSET (SPGradientSelectorClass, released),
-					    gtk_marshal_NONE__NONE,
+					    g_cclosure_marshal_VOID__VOID,
 					    GTK_TYPE_NONE, 0);
 	signals[CHANGED] =  gtk_signal_new ("changed",
 					    (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
 					    GTK_CLASS_TYPE(object_class),
 					    GTK_SIGNAL_OFFSET (SPGradientSelectorClass, changed),
-					    gtk_marshal_NONE__NONE,
+					    g_cclosure_marshal_VOID__VOID,
 					    GTK_TYPE_NONE, 0);
 
 	object_class->destroy = sp_gradient_selector_destroy;
diff -Nurp patched/src/widgets/gradient-vector.cpp working/src/widgets/gradient-vector.cpp
--- patched/src/widgets/gradient-vector.cpp	2009-04-07 22:27:49.000000000 -0700
+++ working/src/widgets/gradient-vector.cpp	2009-04-07 22:28:31.000000000 -0700
@@ -98,7 +98,7 @@ sp_gradient_vector_selector_class_init (
 					      GTK_RUN_LAST,
 					      GTK_CLASS_TYPE(object_class),
 					      GTK_SIGNAL_OFFSET (SPGradientVectorSelectorClass, vector_set),
-					      gtk_marshal_NONE__POINTER,
+					      g_cclosure_marshal_VOID__POINTER,
 					      GTK_TYPE_NONE, 1,
 					      GTK_TYPE_POINTER);
 
diff -Nurp patched/src/widgets/paint-selector.cpp working/src/widgets/paint-selector.cpp
--- patched/src/widgets/paint-selector.cpp	2009-04-07 22:27:49.000000000 -0700
+++ working/src/widgets/paint-selector.cpp	2009-04-07 22:28:31.000000000 -0700
@@ -121,37 +121,37 @@ sp_paint_selector_class_init(SPPaintSele
                                                 (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                                                 GTK_CLASS_TYPE(object_class),
                                                 GTK_SIGNAL_OFFSET(SPPaintSelectorClass, mode_changed),
-                                                gtk_marshal_NONE__UINT,
+                                                g_cclosure_marshal_VOID__UINT,
                                                 GTK_TYPE_NONE, 1, GTK_TYPE_UINT);
     psel_signals[GRABBED] =  gtk_signal_new("grabbed",
                                             (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                                             GTK_CLASS_TYPE(object_class),
                                             GTK_SIGNAL_OFFSET(SPPaintSelectorClass, grabbed),
-                                            gtk_marshal_NONE__NONE,
+                                            g_cclosure_marshal_VOID__VOID,
                                             GTK_TYPE_NONE, 0);
     psel_signals[DRAGGED] =  gtk_signal_new("dragged",
                                             (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                                             GTK_CLASS_TYPE(object_class),
                                             GTK_SIGNAL_OFFSET(SPPaintSelectorClass, dragged),
-                                            gtk_marshal_NONE__NONE,
+                                            g_cclosure_marshal_VOID__VOID,
                                             GTK_TYPE_NONE, 0);
     psel_signals[RELEASED] = gtk_signal_new("released",
                                             (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                                             GTK_CLASS_TYPE(object_class),
                                             GTK_SIGNAL_OFFSET(SPPaintSelectorClass, released),
-                                            gtk_marshal_NONE__NONE,
+                                            g_cclosure_marshal_VOID__VOID,
                                             GTK_TYPE_NONE, 0);
     psel_signals[CHANGED] =  gtk_signal_new("changed",
                                             (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                                             GTK_CLASS_TYPE(object_class),
                                             GTK_SIGNAL_OFFSET(SPPaintSelectorClass, changed),
-                                            gtk_marshal_NONE__NONE,
+                                            g_cclosure_marshal_VOID__VOID,
                                             GTK_TYPE_NONE, 0);
     psel_signals[FILLRULE_CHANGED] = gtk_signal_new("fillrule_changed",
                                                     (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                                                     GTK_CLASS_TYPE(object_class),
                                                     GTK_SIGNAL_OFFSET(SPPaintSelectorClass, fillrule_changed),
-                                                    gtk_marshal_NONE__UINT,
+                                                    g_cclosure_marshal_VOID__UINT,
                                                     GTK_TYPE_NONE, 1, GTK_TYPE_UINT);
 
     object_class->destroy = sp_paint_selector_destroy;
diff -Nurp patched/src/widgets/sp-color-selector.cpp working/src/widgets/sp-color-selector.cpp
--- patched/src/widgets/sp-color-selector.cpp	2009-04-07 22:27:49.000000000 -0700
+++ working/src/widgets/sp-color-selector.cpp	2009-04-07 22:28:31.000000000 -0700
@@ -74,25 +74,25 @@ void sp_color_selector_class_init( SPCol
                                             (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                                             GTK_CLASS_TYPE(object_class),
                                             GTK_SIGNAL_OFFSET(SPColorSelectorClass, grabbed),
-                                            gtk_marshal_NONE__NONE,
+                                            g_cclosure_marshal_VOID__VOID,
                                             GTK_TYPE_NONE, 0 );
     csel_signals[DRAGGED] = gtk_signal_new( "dragged",
                                             (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                                             GTK_CLASS_TYPE(object_class),
                                             GTK_SIGNAL_OFFSET(SPColorSelectorClass, dragged),
-                                            gtk_marshal_NONE__NONE,
+                                            g_cclosure_marshal_VOID__VOID,
                                             GTK_TYPE_NONE, 0 );
     csel_signals[RELEASED] = gtk_signal_new( "released",
                                              (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                                              GTK_CLASS_TYPE(object_class),
                                              GTK_SIGNAL_OFFSET(SPColorSelectorClass, released),
-                                             gtk_marshal_NONE__NONE,
+                                             g_cclosure_marshal_VOID__VOID,
                                              GTK_TYPE_NONE, 0 );
     csel_signals[CHANGED] = gtk_signal_new( "changed",
                                             (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                                             GTK_CLASS_TYPE(object_class),
                                             GTK_SIGNAL_OFFSET(SPColorSelectorClass, changed),
-                                            gtk_marshal_NONE__NONE,
+                                            g_cclosure_marshal_VOID__VOID,
                                             GTK_TYPE_NONE, 0 );
 
     klass->name = nameset;
diff -Nurp patched/src/widgets/sp-color-slider.cpp working/src/widgets/sp-color-slider.cpp
--- patched/src/widgets/sp-color-slider.cpp	2009-04-07 22:27:49.000000000 -0700
+++ working/src/widgets/sp-color-slider.cpp	2009-04-07 22:28:31.000000000 -0700
@@ -88,25 +88,25 @@ sp_color_slider_class_init (SPColorSlide
 						  (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
 						  GTK_CLASS_TYPE(object_class),
 						  GTK_SIGNAL_OFFSET (SPColorSliderClass, grabbed),
-						  gtk_marshal_NONE__NONE,
+						  g_cclosure_marshal_VOID__VOID,
 						  GTK_TYPE_NONE, 0);
 	slider_signals[DRAGGED] = gtk_signal_new ("dragged",
 						  (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
 						  GTK_CLASS_TYPE(object_class),
 						  GTK_SIGNAL_OFFSET (SPColorSliderClass, dragged),
-						  gtk_marshal_NONE__NONE,
+						  g_cclosure_marshal_VOID__VOID,
 						  GTK_TYPE_NONE, 0);
 	slider_signals[RELEASED] = gtk_signal_new ("released",
 						  (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
 						  GTK_CLASS_TYPE(object_class),
 						  GTK_SIGNAL_OFFSET (SPColorSliderClass, released),
-						  gtk_marshal_NONE__NONE,
+						  g_cclosure_marshal_VOID__VOID,
 						  GTK_TYPE_NONE, 0);
 	slider_signals[CHANGED] = gtk_signal_new ("changed",
 						  (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
 						  GTK_CLASS_TYPE(object_class),
 						  GTK_SIGNAL_OFFSET (SPColorSliderClass, changed),
-						  gtk_marshal_NONE__NONE,
+						  g_cclosure_marshal_VOID__VOID,
 						  GTK_TYPE_NONE, 0);
 
 	object_class->destroy = sp_color_slider_destroy;
diff -Nurp patched/src/widgets/sp-color-wheel.cpp working/src/widgets/sp-color-wheel.cpp
--- patched/src/widgets/sp-color-wheel.cpp	2009-04-07 22:27:49.000000000 -0700
+++ working/src/widgets/sp-color-wheel.cpp	2009-04-07 22:28:31.000000000 -0700
@@ -104,7 +104,7 @@ sp_color_wheel_class_init (SPColorWheelC
                           (GtkSignalRunType)(GTK_RUN_FIRST | GTK_RUN_NO_RECURSE),
                           GTK_CLASS_TYPE(object_class),
                           GTK_SIGNAL_OFFSET (SPColorWheelClass, changed),
-                          gtk_marshal_NONE__NONE,
+                          g_cclosure_marshal_VOID__VOID,
                           GTK_TYPE_NONE, 0);
 
     object_class->destroy = sp_color_wheel_destroy;
diff -Nurp patched/src/widgets/sp-widget.cpp working/src/widgets/sp-widget.cpp
--- patched/src/widgets/sp-widget.cpp	2009-04-07 22:27:49.000000000 -0700
+++ working/src/widgets/sp-widget.cpp	2009-04-07 22:28:31.000000000 -0700
@@ -16,6 +16,7 @@
 #include "macros.h"
 #include "../document.h"
 #include "sp-widget.h"
+#include "helper/sp-marshal.h"
 
 enum {
 	CONSTRUCT,
@@ -78,27 +79,27 @@ sp_widget_class_init (SPWidgetClass *kla
 						    GTK_RUN_FIRST,
 						    GTK_CLASS_TYPE(object_class),
 						    GTK_SIGNAL_OFFSET (SPWidgetClass, construct),
-						    gtk_marshal_NONE__NONE,
+						    g_cclosure_marshal_VOID__VOID,
 						    GTK_TYPE_NONE, 0);
 	signals[CHANGE_SELECTION] = gtk_signal_new ("change_selection",
 						    GTK_RUN_FIRST,
 						    GTK_CLASS_TYPE(object_class),
 						    GTK_SIGNAL_OFFSET (SPWidgetClass, change_selection),
-						    gtk_marshal_NONE__POINTER,
+						    g_cclosure_marshal_VOID__POINTER,
 						    GTK_TYPE_NONE, 1,
 						    GTK_TYPE_POINTER);
 	signals[MODIFY_SELECTION] = gtk_signal_new ("modify_selection",
 						    GTK_RUN_FIRST,
 						    GTK_CLASS_TYPE(object_class),
 						    GTK_SIGNAL_OFFSET (SPWidgetClass, modify_selection),
-						    gtk_marshal_NONE__POINTER_UINT,
+						    sp_marshal_VOID__POINTER_UINT,
 						    GTK_TYPE_NONE, 2,
 						    GTK_TYPE_POINTER, GTK_TYPE_UINT);
 	signals[SET_SELECTION] =    gtk_signal_new ("set_selection",
 						    GTK_RUN_FIRST,
 						    GTK_CLASS_TYPE(object_class),
 						    GTK_SIGNAL_OFFSET (SPWidgetClass, set_selection),
-						    gtk_marshal_NONE__POINTER,
+						    g_cclosure_marshal_VOID__POINTER,
 						    GTK_TYPE_NONE, 1,
 						    GTK_TYPE_POINTER);
 
